lumigator/frontend/src/components/views/Settings.vue

```diff
@@ -0,0 +1,354 @@
+                v-model="reference.value"
+                aria-describedby="api-key-label"
+                :placeholder="`${title} API Key`"
+              />
+              <Button
+                class="delete-button button"
+                icon="pi pi-trash"
+                @click="deleteSecret(apiKey, title)"
+                v-if="existsRemotely"
+              ></Button>
+            </div>
+            <Button
+              class="save-button button"
+              @click="
+                uploadSecret(
+                  {
+                    name: apiKey,
+                    description: `${title} API Key`,
+                    value: reference.value,
+                  },
+                  title,
+                )
+              "
+              :disabled="!isValidApiKey(reference.value)"
+              label="Save"
+            ></Button>
+          </div>
+        </div>
+      </div>
+    </div>
+  </div>
+</template>
+
+<script setup lang="ts">
+import { settingsService } from '@/sdk/settingsService'
+import type { SecretUploadPayload } from '@/types/Secret'
+import { InputText, type ToastMessageOptions } from 'primevue'
+import Button from 'primevue/button'
+import { onMounted, ref, type Ref } from 'vue'
+import { useConfirm } from 'primevue/useconfirm'
+import { useToast } from 'primevue/usetoast'
+import axios, { AxiosError } from 'axios'
+import { getAxiosError } from '@/helpers/getAxiosError'
+
+const confirm = useConfirm()
+const toast = useToast()
+// Placeholder for configured secrets where the actual value is hidden.
+const maskedValue = '****************'
+
+// API key map is used to track API key names (based on the provider name) to their corresponding ref, title and whether the setting exists remotely.
+const apiKeyMap = new Map<
+  string,
+  { reference: Ref<string>; existsRemotely: boolean; title: string }
+>([
+  ['mistral_api_key', { reference: ref(''), existsRemotely: false, title: 'Mistral' }],
+  ['openai_api_key', { reference: ref(''), existsRemotely: false, title: 'OpenAI' }],
+  ['hf_api_key', { reference: ref(''), existsRemotely: false, title: 'Hugging Face' }],
+  ['deepseek_api_key', { reference: ref(''), existsRemotely: false, title: 'DeepSeek' }],
+])
+
+onMounted(async () => {
+  fetchSecrets()
+})
+
+// Check if the API key is valid (e.g. it is some characters, but not the masked value).
+const isValidApiKey = (value: string): boolean => {
+  return value !== maskedValue && value.length > 0
+}
+
+const fetchSecrets = async () => {
+  try {
+    const secrets = await settingsService.fetchSecrets()
+    apiKeyMap.forEach((obj, secretKey) => {
+      obj.existsRemotely = secrets.some((secret) => secret.name == secretKey)
+      obj.reference.value = obj.existsRemotely ? maskedValue : ''
+    })
+  } catch (error) {
+    const msg = extractErrorMessages(error).join(', ')
+    toast.add({
+      severity: 'error',
+      summary: 'Unable to fetch configured API keys',
+      detail: `${msg}`,
```

Comment:
nit: no need for the template strings if we're just passing the variable

```suggestion
      detail: msg,
```